### PR TITLE
Enhancement: Enable Hermes engine by default

### PR DESCRIPTION
**Pull request by frankcalise**
## Please verify the following:

- [x] `README.md` has been updated with your changes, if relevant

## Describe your PR

Enables Hermes JS engine by default.

|    `Build`     |   `Hermes`    | 
| ------------------------------- | ------------------------------- |
| iOS | ✅ |
| Android | ✅  |
| Expo Android | Expo Go, expo-dev-client, EAS Build  |
| Expo iOS* | expo-dev-client, EAS Build |

* Note: Hermes is *not* available Expo Go for iOS

### CODE REVIEW

File: boilerplate/ios/Podfile [hunk 1]
```diff
@@ -27,7 +27,7 @@ target 'HelloWorld' do
     :path => config[:reactNativePath],
     # to enable hermes on iOS, change `false` to `true` and then install pods
     :production => production,
-    :hermes_enabled => flags[:hermes_enabled],
+    :hermes_enabled => true,
```
Comment (jamonholmgren) [inline boilerplate/ios/Podfile:30]: Does `flags[:hermes_enabled]` not work here?

**Review by jamonholmgren — APPROVED** [review]

File: boilerplate/ios/Podfile (hunk 1 above)
Comment (frankcalise) [reply boilerplate/ios/Podfile:30]: AFAIK enabling the new arch flag will turn everything on - https://github.com/react-native-community/upgrade-support/issues/193#issuecomment-1158371500

I'm not familiar enough with pod files, is there a better way to pass this along on the pod install command line?

**Comment by jamonholmgren** [comment]
I think this is ready for review, and probably (other than my one comment) good to go.

File: boilerplate/ios/Podfile (hunk 1 above)
Comment (jamonholmgren) [reply boilerplate/ios/Podfile:30]: I don't know, tbh. I'm okay with this, maybe with a comment saying we might want to move back to `flags[:hermes_enabled]` in the future?

**Pull request merged**